mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5181,6 +5181,19 @@ std::optional<SmallVector<int64_t, 4>> GatherOp::getShapeForUnroll() {
   return llvm::to_vector<4>(getVectorType().getShape());
 }
 
+static LogicalResult isContiguousIndices(Value val) {
```

Comment:
Could you add doc about the current supported cases and limitations.

nit: `val` -> `indices`, `indexVec` ... ?